### DISCUSSION TITLE
Git workflow for editing pipeline more cleanly

### DISCUSSION THREAD

**Discussion by galengorski** [USGS-R/drb-estuary-salinity-ml#91] [General]
This is a generalized workflow for editing different parts of the pipeline more cleanly so that we can try to avoid multiple people working on the same part of the project. Also it's a place where I can put these git commands so I don't forget them. 

- Working on `03b_model/src/run_model.py` on a branch called `model_edits` and find that a change to `01_fetch/src/fetch_usgs_nwis.py` is required
- on `model_edits` commit changes to `run_model.py`
- switch back to main branch on local
- create new branch (from the main branch) for small changes to `fetch_usgs_nwis.py` using `git checkout -b fetch_changes`
- make small changes to fetch, commit, `push upstream fetch_changes` creating new branch in remote repo
- merge small changes to fetch into remote repo main branch
- back on local, switch to main branch `git checkout main`
- pull the changes from remote repo in local main using `git pull upstream main`
- switch back to `model_edits` branch
- merge in new changes `git merge main -m "merge small change to fetch nwis"`

**Comment by amsnyder** [comment]
Works for me! Thanks for writing this up @galengorski
